### PR TITLE
Lesson 03 task

### PR DESCRIPTION
**Pull request by karamel29**

### CODE REVIEW

File: task003/main.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+// function that returns the copy of the original slice in reverse order
+func reverse(numbers []int64) []int64 {
+	for i := 0; i < len(numbers)/2; i++ {
+		j := len(numbers) - i - 1
+		numbers[i], numbers[j] = numbers[j], numbers[i]
+	}
+	return numbers
+}
```
Comment (IgorZhadenovEpam) [inline task003/main.go]: There is a little problem with this code
"Implement reverse([]int64) []int64 function, that returns the **COPY** of the original slice in reverse order. "
But you reverse and return the original slice 
https://play.golang.org/p/e4pgoTKTPbz

File: task003/main.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	}
+	n := sum / float64(len(s))
+	return n
+}
+
+// function that returns the longest word from the slice of strings
+func maxString(words []string) string {
+	var maxWord string
+	for _, word := range words {
+		if len(word) > len(maxWord) {
```
Comment (DzmitryYafremenka) [inline task003/main.go]: Input words1 := []string{"从", "too", "be", "or"}
Returns the result "从". This is 1 character. The correct result should be "too".

File: task003/main.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+	words2 := []string{"one", "two"}
+	maxWord1 := maxString(words2)
+	fmt.Println("The longest word of two equals:", maxWord1)
+
+	r := []int64{1, 2, 5, 15}
+	n := reverse(r)
+	fmt.Println("The copy of the original slice in reverse order:", n)
+
+	key1 := map[int]string{2: "a", 0: "b", 1: "c"}
```
Comment (DzmitryYafremenka) [inline task003/main.go:73]: If we try to add
`fmt.Println(key1)` right after this one the output will be:
```
map[0:b 1:c 2:a]
Map values sorted in order of increasing keys: [b c a]
```
same order as in sorted map.  So why do we need to sort map?
Assuming code is right - try to figure out why fmt.Println() outputs a sorted list and how to output an unsorted map.

p.s. no need to fix this code.

**Review by DzmitryYafremenka — CHANGES_REQUESTED** [review]

File: task003/main.go (hunk 3 above)
Comment (karamel29) [reply task003/main.go:73]: A map is an unordered collection of key-value pairs.
If you need a stable iteration order, you must maintain a separate data structure.

Also, starting with Go 1.12, the fmt package prints maps in key-sorted order to ease testing. So, this [article](https://habr.com/ru/post/457728/) says: "Причину можно найти в исходном коде рантайма языка:

```
// mapiterinit initializes the hiter struct used for ranging over maps. 
func mapiterinit(t *maptype, h *hmap, it *hiter) {...
// decide where to start 
r := uintptr(fastrand())
 ... 
it.startBucket = r & bucketMask(h.B)...}
```

Место поиска определяется рандомно, запомните это! Ходят слухи, что так разработчики рантайма заставляют пользователей не полагаться на порядок."

> How to output an unsorted map

Idk the answer, but judging by the information I found, no way. Or is there some way to do it?

Comment (DzmitryYafremenka) [reply task003/main.go:73]: Yes, there is a way. All the randomness of the map is exposed when you iterate over it. Check out this Daft Punk song generator here as an example.
https://play.golang.org/p/7-vgnDpmd4b

**Review by IgorZhadenovEpam — APPROVED** [review]

File: task003/main.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	}
+	n := sum / float64(len(s))
+	return n
+}
+
+// function that returns the longest word from the slice of strings
+func maxString(words []string) string {
+	var maxWord string
+	for _, word := range words {
+	 if utf8.RuneCountInString(word) > utf8.RuneCountInString(maxWord) {
```
Comment (AliakseiM) [inline task003/main.go:24]: Nice!

**Review by AliakseiM — APPROVED** [review]

**Review by DzmitryYafremenka — APPROVED** [review]

**Pull request merged**